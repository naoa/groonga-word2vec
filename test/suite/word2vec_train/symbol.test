plugin_register word2vec/word2vec

table_create Tags TABLE_PAT_KEY ShortText

table_create Entries TABLE_NO_KEY
column_create Entries title COLUMN_SCALAR ShortText
column_create Entries tag COLUMN_SCALAR Tags
column_create Entries tags COLUMN_VECTOR Tags

load --table Entries
[
{"title": "今日は＊雨だ1\",.;:&^/-#'()なぁ", "tag": "Library"},
{"title": "明日は<b>晴れ</b>だなぁ", "tag": "Server"}
]

word2vec_train Entries title$,tag --min_count 1
word2vec_distance "Server" --limit 40
