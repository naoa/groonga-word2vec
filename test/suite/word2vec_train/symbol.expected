plugin_register word2vec/word2vec
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Entries tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
column_create Entries tags COLUMN_VECTOR Tags
[[0,0.0,0.0],true]
load --table Entries
[
{"title": "今日は＊雨だ1\",.;:&^/-#'()[]なぁ", "tag": "Library"},
{"title": "明日は<b>晴れ</b>だなぁ", "tag": "Server"}
]
[[0,0.0,0.0],2]
word2vec_train Entries title$,tag --min_count 1
[[0,0.0,0.0],true]
word2vec_distance "Server" --limit 40
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      7
    ],
    [
      [
        "_key",
        "ShortText"
      ],
      [
        "_value",
        "Float"
      ]
    ],
    [
      "なぁ",
      0.236046344041824
    ],
    [
      "</s>",
      0.121626138687134
    ],
    [
      "今日は*雨だ",
      0.0758075639605522
    ],
    [
      "だなぁ",
      0.00604319386184216
    ],
    [
      "明日は",
      -0.0134698543697596
    ],
    [
      "library",
      -0.0158039312809706
    ],
    [
      "晴れ",
      -0.0652221590280533
    ]
  ]
]
